### ISSUE TITLE
One of the variables needed for gradient computation has been modified by an inplace operation

### ISSUE THREAD

**Issue by jimmy-dq** [Kim-Minseon/RoCL#4]
Hi,
Thanks for your great work. I ran rocl_train.py with your default parameters and find one error output when executing `loss.backward()`, the error is:
`_RuntimeError: one of the variables needed for gradient computation has been modified by an inplace operation.`
I found the problem is in `adv_loss`. After we execute `outputs = projector(model(inputs))`, `adv_loss` cannot be backwarded due to the inplace operation. Appreciate for your help.

**Comment by immuntasir** [comment]
Hi, were you able to solve this?

**Comment by wangren09** [comment]
Hi, I met the same problem.

**Comment by chlorophyllccc** [comment]
Hi, I met the same problem, do you solve this?

**Comment by cfeng16** [comment]
Hi, I met the same problem, Have you solved it?

**Comment by ramithuh** [comment]
@immuntasir were you able to solve it ? 😃

**Comment by priyamdey** [comment]
It runs fine with multiple gpus. Error pops up for single gpu run. Comment out the lines [88](https://github.com/Kim-Minseon/RoCL/blob/b6d5185e294e8bca5670e9146df81b54d99d0635/src/rocl_train.py#L88) to [100](https://github.com/Kim-Minseon/RoCL/blob/b6d5185e294e8bca5670e9146df81b54d99d0635/src/rocl_train.py#L100) which wraps the model with SyncBN & DDP. Also the init (line [47](https://github.com/Kim-Minseon/RoCL/blob/b6d5185e294e8bca5670e9146df81b54d99d0635/src/rocl_train.py#L47) to [52](https://github.com/Kim-Minseon/RoCL/blob/b6d5185e294e8bca5670e9146df81b54d99d0635/src/rocl_train.py#L52), although I haven't tried with keeping this intact). Accordingly, update the run command by removing the distributed part. It should work fine then.
